frontend/src/components/admin/assessment-creation/AssessmentPreview.tsx

```diff
@@ -1,18 +1,32 @@
-import React, { useContext } from "react";
+import type { ReactNode } from "react";
+import React, { useEffect } from "react";
 import { Button } from "@chakra-ui/react";
 
 import { EditOutlineIcon } from "../../../assets/icons";
-import AssessmentContext from "../../../contexts/AssessmentContext";
+import type { Question } from "../../../types/QuestionTypes";
 import AssessmentExperience from "../../student/AssessmentExperience";
 
-const AssessmentPreview = (): React.ReactElement => {
-  const { questions, setShowAssessmentPreview } = useContext(AssessmentContext);
+type AssessmentPreviewProps = {
+  questions: Question[];
+  goBack: () => void;
+};
+
+const AssessmentPreview = ({
+  questions,
+  goBack,
+}: AssessmentPreviewProps): ReactNode => {
+  useEffect(() => {
+    if (!questions.length) {
+      goBack();
+    }
+  }, [goBack, questions]);
+
+  if (!questions.length) {
+    return null;
+  }
```

Comment:
is this necessary? since we already validate form before preview so questions will never be empty

Reply:
Yeah, the page is now loadable directly by URL so we need validation in that case (e.g., invalid assessment ID)

Reply:
makes sense